web/src/refresh-components/inputs/InputComboBox/InputComboBox.tsx

```diff
@@ -0,0 +1,397 @@
+      <div className="relative">
+        <InputTypeIn
+          ref={inputRef}
+          placeholder={placeholder}
+          value={displayLabel}
+          onChange={handleInputChange}
+          onFocus={handleFocus}
+          onKeyDown={handleKeyDown}
+          disabled={disabled}
+          isError={!isValid}
+          leftSearchIcon={leftSearchIcon}
+          showClearButton={false}
+          rightSection={
+            <>
+              {rightSection && (
+                <div
+                  className="flex items-center"
+                  onPointerDown={(e) => {
+                    e.stopPropagation();
+                  }}
+                  onClick={(e) => {
+                    e.stopPropagation();
+                  }}
+                >
+                  {rightSection}
+                </div>
+              )}
+              {hasOptions && (
+                <IconButton
+                  internal
+                  onClick={(e) => {
+                    e.stopPropagation();
+                    toggleDropdown();
+                  }}
+                  disabled={disabled}
+                  icon={isOpen ? SvgChevronUp : SvgChevronDown}
+                  aria-label={isOpen ? "Close dropdown" : "Open dropdown"}
+                  tabIndex={-1}
+                  type="button"
+                />
+              )}
+            </>
+          }
+          {...ariaProps}
+          {...rest}
+        />
+
+        {/* Dropdown - Rendered in Portal */}
+        <ComboBoxDropdown
+          ref={dropdownRef}
+          isOpen={isOpen}
+          disabled={disabled}
+          dropdownPosition={dropdownPosition}
+          fieldId={fieldId}
+          placeholder={placeholder}
+          matchedOptions={matchedOptions}
+          unmatchedOptions={unmatchedOptions}
+          hasSearchTerm={hasSearchTerm}
+          separatorLabel={separatorLabel}
+          value={value}
+          highlightedIndex={highlightedIndex}
+          onSelect={handleOptionSelect}
+          onMouseEnter={(index) => {
+            setIsKeyboardNav(false);
+            setHighlightedIndex(index);
+          }}
+          onMouseMove={() => {
+            if (isKeyboardNav) {
+              setIsKeyboardNav(false);
+            }
+          }}
+          isExactMatch={isExactMatch}
+        />
+      </div>
+
+      {/* Error message - only show internal error messages when not using external isError */}
+      {!isValid && errorMessage && externalIsError === undefined && (
+        <div className="flex flex-row items-center gap-x-0.5 ml-0.5 mt-1">
+          <div className="w-4 h-4 flex items-center justify-center">
+            <SvgXOctagon className="h-3 w-3 stroke-status-error-05" />
+          </div>
+          <Text
+            id={`${fieldId}-error`}
+            text03
+            secondaryBody
+            className="ml-0.5"
+            role="alert"
+          >
+            {errorMessage}
+          </Text>
+        </div>
```

Comment:
This may be a good place to extract "error message" chips out into their own component.